lib/pf/Switch/Hostapd.pm

```diff
@@ -177,6 +177,44 @@ sub radiusDisconnect {
     return;
 }
 
+=item returnRadiusAccessAccept
+
+Prepares the RADIUS Access-Accept reponse for the network device.
+
+=cut
+
+
+sub returnRadiusAccessAccept {
+    my ($self, $args) = @_;
+    my $logger = $self->logger;
+
+    $args->{'unfiltered'} = $TRUE;
+    my @super_reply = @{$self->SUPER::returnRadiusAccessAccept($args)};
+    my $status = shift @super_reply;
+    my %radius_reply = @super_reply;
+    my $radius_reply_ref = \%radius_reply;
+    return [$status, %$radius_reply_ref] if($status == $RADIUS::RLM_MODULE_USERLOCK);
+
+    if (isenabled($args->{profile}->{_ipsk})) {
```

Comment:
I think this logic should be wrapped in something like "getUserPSK" or something like that

Then if its defined, you put it in the necessary attribute

That way we don't need to duplicate the "enabled" logic and the logic to find the default key if applicable